src/Microsoft.AspNet.Security.OAuth/OAuthAuthenticationHandler.cs

```diff
@@ -176,7 +171,7 @@ protected override AuthenticationTicket AuthenticateCore()
                 Properties = properties,
             };
             await Options.Notifications.GetUserInformationAsync(context);
-            return new AuthenticationTicket(context.Identity, context.Properties);
+            return new AuthenticationTicket(context.Principal, context.Properties, Options.AuthenticationScheme);
         }
 
         protected override void ApplyResponseChallenge()
```

Comment:
On a separate note: ExchangeCodeAsync has no extensibility for adding parameters when exchanging a code for an AccessToken. Extensibility is required for a lot of customers. We are currently designing a 'code' flow including token / refresh caching. We will share the design.

Just a heads up to avoid duplication of work. If any other changes are planned for OAuthMW we should have a conversation.